x-pack/plugins/security_solution/common/endpoint/types/sentinel_one.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * The `activity` document ingested from SentinelOne via the integration
+ *
+ * NOTE:  not all properties are currently mapped below. Check the index definition if wanting to
+ *        see what else is available and add it bellow if needed
+ */
+export interface SentinelOneActivityEsDoc {
+  sentinel_one: {
+    activity: {
+      agent: {
+        /** This is the internal ID of the host in sentinelOne (NOT the agent's UUID) */
+        id: string;
+      };
+      updated_at: string;
+      description: {
+        primary: string;
+        secondary?: string;
+      };
+      id: string;
+      /** The activity type. Valid values can be retrieved from S1 via API: `/web/api/v2.1/activities/types` */
+      type: number;
+    };
+  };
+}
+
+export interface SentinelOneActionRequestCommonMeta {
+  /** The S1 agent id */
+  agentId: string;
+  /** The S1 agent assigned UUID */
```

Comment:
I have trouble understanding what is the difference between `agentId` and `agentUUID` - could you explain in more details?

Reply:
I guess one is referencing Elastic Agent with the integration?

Reply:
Not sure its the "official" answer, but:

In SentinelOne, agents have both an `id` and a `uuid`. It seems to me that the `uuid` can be re-generated - I think I saw some action on the agent there to re-assign. new UUID.   The `id` seems to be more in line with how we use "id" in elastic and this also what is used internally in SentinelOne to trigger several actions or run scripts.

Because we are using the UUID of the agent in kibana we need to (in this case) store the (what I am calling "internal") `id` of the agent in SentinelOne with the action so that we can use it later without having to call their API again.